File: src/main/java/rk/cinema/model/Movie.java
package rk.cinema.model;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import jakarta.validation.constraints.NotBlank;

@MappedEntity
public record Movie(
        @Id
        @GeneratedValue
        Long id,

        @NonNull
        @NotBlank
        String title,

        @NonNull
        Integer durationInMinutes // czas trwania
) {
}

File: src/main/kotlin/rk/cinema/model/Movie.kt
package rk.cinema.model

import io.micronaut.core.annotation.NonNull
import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import jakarta.validation.constraints.NotBlank

@MappedEntity
data class Movie(
    @GeneratedValue @param:Id val id: Long,
    val title: @NonNull @NotBlank String,
    val durationInMinutes: @NonNull Int,
)

